stl/inc/algorithm

```diff
@@ -6067,6 +6068,209 @@ _BidIt stable_partition(_ExPo&&, _BidIt _First, _BidIt _Last, _Pr _Pred) noexcep
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::stable_partition
+    template <bidirectional_iterator _It>
+    _It _Buffered_rotate_common(const _It _First, const _It _Mid, const _It _Last, const iter_difference_t<_It> _Count1,
+        const iter_difference_t<_It> _Count2, iter_value_t<_It>* const _Temp_ptr, const ptrdiff_t _Capacity) {
+        // rotate [_First, _Last) using temp buffer
+        // precondition: _Count1 == distance(_First, _Mid)
+        // precondition: _Count2 == distance(_Mid, _Last)
+        if (_Count1 == 0) {
+            return _Last;
+        }
+
+        if (_Count2 == 0) {
+            return _First;
+        }
+
+        if (_Count1 <= _Count2 && _Count1 <= _Capacity) { // buffer left range, then copy parts
+            _Uninitialized_backout<iter_value_t<_It>*> _Backout{
+                _Temp_ptr, _RANGES _Uninitialized_move_unchecked(_First, _Mid, _Temp_ptr, _Temp_ptr + _Count1).out};

```

Comment:
I really do not like this function call as function argument.

I will most likely go over this soonish and give the result proper names